app/jobs/shipit/perform_task_job.rb

```diff
@@ -62,6 +63,10 @@ def perform_task
 
     def checkout_repository
       unless @commands.fetched?(@task.until_commit).tap(&:run).success?
+        # acquire_git_cache_lock can take upto 15 seconds
+        # to process. Try to make sure that the job isn't
+        # marked dead while we attempt to acquire the lock.
+        @task.ping
         @task.acquire_git_cache_lock do
```

Comment:
We should also probably ping again after we got the lock.

Reply:
I don't have strong feelings on renewing the "alive" key after acquiring the lock. I had considered it, but decided against it because it seemed like it might be overkill. My reasoning: 

1. I've upped the lifetime of the "alive" key in redis to 30 seconds. 
2. Acquiring the lock times out after 15 seconds. I imagined that it would take far less than the remaining 15 seconds on the "alive" key to execute `@commands.fetched?(@task.until_commit).tap(&:run).success?` 
3. The very next thing we do is reset the "alive" key countdown as a side effect of `capture! @commands.fetch`. 

Do you think it'd be worth it to add here anyway?

Reply:
Yeah, I'd just rather be safe than sorry.

Reply:
Addressed by: https://github.com/Shopify/shipit-engine/pull/1055/commits/da1a05e148667774fbb292b2a6ea11a1ed4167f0#diff-8f7af914ee8af7005ec9a5585f1d848eR71